### PR TITLE
CLN: remove unused min_count argument in libgroupby.group_nth

### PR DESCRIPTION
**Pull request by arw2019**
- [ ] closes #xxxx
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by WillAyd** [comment]
Does this have any impact to an end user?

**Comment by arw2019** [comment]
> Does this have any impact to an end user?

Good point. I *think* it allows the user to pass `min_count` to `group_nth`. Will see if I can come up with a test

File: pandas/core/groupby/ops.py [hunk 1]
```diff
@@ -603,8 +603,7 @@ def _aggregate(
     ):
         if agg_func is libgroupby.group_nth:
             # different signature from the others
-            # TODO: should we be using min_count instead of hard-coding it?
-            agg_func(result, counts, values, comp_ids, rank=1, min_count=-1)
```
Comment (jbrockmendel) [inline pandas/core/groupby/ops.py:607]: one alternative here is to remove the keyword from the libgroupby function.

Comment (arw2019) [reply pandas/core/groupby/ops.py:607]: I went with that

Actually we don't use the min_count argument in the libgroupby function - I guess that it was put there to homogenize signatures, but now they're different anyways

**Comment by jreback** [comment]
thanks @arw2019

**Pull request merged**